### DISCUSSION TITLE
Contenteditable - Component proposal

### DISCUSSION THREAD

**Discussion by ChrisUser** [shadcn-ui/ui#4250] [Ideas]
<div align="center">

![react-contenteditable](https://github.com/ChrisUser/react-basic-contenteditable/raw/main/readme-header-img.png)

</div>

---

Hello everyone 👋 (devs and consumers),

I wanted to ask you what you think of the idea of ​​integrating my [react-basic-contenteditable](https://github.com/ChrisUser/react-basic-contenteditable) among the shadcn ui components.
I noticed that the official list does not have it and since it is a type of component that is used in numerous interfaces lately (because of AI integrations almost everywhere) I thought of creating one that is entirely monitorable and customizable.

<div align="center">

![contenteditable](https://github.com/shadcn-ui/ui/assets/61348788/eee40687-113b-42d7-9b1d-32dc92d8be58)

</div>

The custom style can be added by using the three `*ClassName` props that accept string such as basic css classes or utility classes as in the case of Tailwind.

> Why not using the simple input component?

This one is a multiline dynamic div with customizable dimensions and simple controls made exactly for interfaces that require more than just a text input, such as messaging apps or AI prompt ones. You can try it [here](https://chrisuser.github.io/react-basic-contenteditable/).

❗ **This post is not just to show off, but a simple request to understand if it's a component/functionality with a higher demand.** ❗

Please, let me know what you think and also if it's possible to integrate my external component in this library like they did with "Sonner". If there are steps or a checklist to follow for that to be made I would be interested to know that.

**Comment by yantakus** [comment]
Would be great to have it.

**Comment by ChrisUser** [comment]
Thanks for all the positive feedback!
As I said before, I wasn't able to find steps to "officially ask" or a checklist to follow for it to be made into a legacy component of shadcn-ui.

Maybe if enough people will be interested in this proposal the mantainers might reach out?
